GameTheoryLanguage/Visitor/CustomGtlVisitor.cs

```diff
@@ -283,45 +363,42 @@ public override object VisitArgCallExpr(GtlParser.ArgCallExprContext context)
         string[][] functionTypes = Ftable.Find(functionId);
         if (stringArray.Length != functionTypes[0].Length)
         {
-            throw new NotSupportedException("Not the correct amount of input parameters in function");
+            throw new FunctionCallException("Not the correct amount of input parameters in function");
         }
         for (int i = 0; i < stringArray.Length; i++)
         {
             if (!stringArray[i].Equals(functionTypes[0][i]))
             {
-                throw new NotSupportedException($"expected type {functionTypes[0][i]} in function call but recieved {stringArray[i]}");
+                throw new FunctionCallException($"expected type {functionTypes[0][i]} in function call but received {stringArray[i]}");
             }
         }
         return Ftable.Find(functionId)[1][0];
     }
-    /*
     public override object VisitParExpr(GtlParser.ParExprContext context)
     {
-        visit expression and return its returntype
+        Console.WriteLine("Visting Parentheses expression");
+        return (string)Visit(context.expr());
     }
+    /*
     public override object VisitUtilExpr(GtlParser.UtilExprContext context)
     {
         ??????
     }
+    */
     public override object VisitTupleExpr(GtlParser.TupleExprContext context)
     {
-        ?????
+        Console.WriteLine("visiting tuple expression");
+        return (string)Visit(context.tuple());
```

Comment:
Her kan jeg ikke se hvorfor det skal være Visit(context...) i stedet for base.VisitTupl....